include/arch/riscv/arch/object/objecttype.h

```diff
@@ -43,5 +43,42 @@ static inline CONST bool_t Arch_isIRQControlDescendant(cap_t cap)
  */
 static inline CONST bool_t Arch_isMDBParentOf(cap_t cap_a, cap_t cap_b, bool_t firstBadged)
 {
-    return true;
+    switch (cap_get_capType(cap_a)) {
+#ifdef CONFIG_ALLOW_SBI_CALLS
+    case cap_sbi_cap: {
+        if (!cap_sbi_cap_get_capSBIEIDBadged(cap_a)) {
+            /* If there is no EID badge then the cap cannot be badged. */
+            return true;
```

Comment:
This comment is very misleading, it makes it sound like false should be returned.

I guess what was meant is that if cap A has no badge, cap B must be either a copy or a badged version of A.